django/thunderstore/api/cyberstorm/views/packages.py

```diff
@@ -0,0 +1,467 @@
+            )
+
+        return queryset
+
+    def _filter_by_section(
+        self,
+        section_slug: Optional[str],
+        queryset: QuerySet[Package],
+    ) -> QuerySet[Package]:
+        """
+        PackageListingSections can be used as shortcut for multiple
+        category filters.
+        """
+        if not section_slug:
+            return queryset
+
+        try:
+            section = PackageListingSection.objects.prefetch_related(
+                "require_categories",
+                "exclude_categories",
+            ).get(slug=section_slug)
+        except PackageListingSection.DoesNotExist:
+            required = []
+            excluded = []
+        else:
+            required = section.require_categories.values_list("pk", flat=True)
+            excluded = section.exclude_categories.values_list("pk", flat=True)
+
+        if required:
+            queryset = queryset.exclude(
+                ~Q(community_listings__categories__pk__in=required),
+            )
+
+        if excluded:
+            queryset = queryset.exclude(community_listings__categories__pk__in=excluded)
+
+        return queryset
+
+    def _filter_by_query(
+        self,
+        query: Optional[str],
+        queryset: QuerySet[Package],
+    ) -> QuerySet[Package]:
+        """
+        Filter packages by free text search.
+        """
+        if not query:
+            return queryset
+
+        search_fields = ("name", "owner__name", "latest__description")
+        icontains_query = Q()
+        parts = [x for x in query.split(" ") if x]
+
+        for part in parts:
+            for field in search_fields:
+                icontains_query &= ~Q(**{f"{field}__icontains": part})
+
+        return queryset.exclude(icontains_query).distinct()
+
+    def _order_queryset(
+        self,
+        ordering: str,
+        queryset: QuerySet[Package],
+    ) -> QuerySet[Package]:
+        """
+        Order results in requested order, defaulting to latest update.
+        """
+        order_arg = {
+            "last-updated": "-date_updated",
+            "most-downloaded": "-download_count",
+            "newest": "-date_created",
+            "top-rated": "-rating_count",
+        }[ordering]
+
+        return queryset.order_by("-is_pinned", "is_deprecated", order_arg)
+
+    def _paginate(self, params: OrderedDict, queryset: QuerySet[Package]) -> Page:
+        """
+        Slice queryset based on the requested page.
+        """
+        paginator = CachedPaginator(
+            queryset,
+            self.page_size,
+            cache_key=self._get_paginator_cache_key(),
+            cache_vary=self._get_full_cache_vary(params),
+            cache_bust_condition=CacheBustCondition.any_package_updated,
+        )
+
+        # PageNotAnInteger error won't be raised here since deserializer
+        # has already checked that the page parameter is an integer.
```

Comment:
No clue if `_paginate` is a custom function or an override from DRF, but assuming it's the former, I'd prefer type-annotating a `page` parameter as an integer to this function & relying on that instead of commenting on how the params _should_ be already validated to be the correct format. From the perspective of this function we have no clue what actually is passed through `params` after all and now we need to think about a wider context to really get a grasp of what's going on here.

Not a blocker

Reply:
Ensured the comment won't get outdated by removing it altogether.